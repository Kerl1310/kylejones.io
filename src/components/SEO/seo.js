import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { StaticQuery, graphql } from 'gatsby';

function SEO({ description, lang, meta, keywords, title, featuredImage, url }) {
  return (
    <StaticQuery
      query={detailsQuery}
      render={data => {
        const metaDescription =
          description || data.site.siteMetadata.description;
        return (
          <React.Fragment><script type="application/ld+json"> {`
          {
            "@context": "https://schema.org",
            "@type": "Author",
            "birthDate": "1992-10-13",
            "email": "kylejones1310@outlook.com",
            "gender": "Male",
            "jobTitle": "Senior Software Engineer",
            "knowsLanguage": "English",
            "name": "Kyle Jones",
            "telephone": "07432 829513",
            "url": "https://kylejones.io"
            "name": "Kyle Jones",
          }
        `}
        </script><Helmet
            htmlAttributes={{
              lang,
            }}
            title={title}
            titleTemplate={`${data.site.siteMetadata.author}`}
            meta={[
              {
                name: `description`,
                content: metaDescription,
              },
              {
                property: `og:title`,
                content: title,
              },
              {
                property: `og:url`,
                content: url,
              },
              {
                property: `og:description`,
                content: metaDescription,
              },
              {
                property: `og:locale`,
                content: `en-GB`,
              },
              {
                property: `og:type`,
                content: `website`,
              },
              {
                property: `og:image`,
                content: featuredImage,
              },
              {
                name: `twitter:card`,
                content: `summary`,
              },
              {
                name: `twitter:creator`,
                content: data.site.siteMetadata.author,
              },
              {
                name: `twitter:title`,
                content: title,
              },
              {
                name: `twitter:site`,
                content: data.site.siteMetadata.twitter,
              },
              {
                name: `twitter:image`,
                content: featuredImage,
              },
              {
                name: `twitter:domain`,
                content: data.site.siteMetadata.siteUrl,
              },
              {
                name: `twitter:description`,
                content: metaDescription,
              },
            ]
              .concat(
                keywords.length > 0
                  ? {
                      name: `keywords`,
                      content: keywords.join(`, `),
                    }
                  : []
              )
              .concat(meta)}
          /></React.Fragment>
        )
      }}
    />
  );
}

SEO.defaultProps = {
  lang: `en`,
  meta: [],
  keywords: [],
};

SEO.propTypes = {
  description: PropTypes.string,
  lang: PropTypes.string,
  meta: PropTypes.array,
  keywords: PropTypes.arrayOf(PropTypes.string),
  title: PropTypes.string.isRequired,
};

export default SEO;

const detailsQuery = graphql`
  query DefaultSEOQuery {
    site {
      siteMetadata {
        title
        description
        author
      }
    }
  }
`;
