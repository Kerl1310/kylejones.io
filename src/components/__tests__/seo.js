import React from 'react';
import renderer from 'react-test-renderer';
import { StaticQuery } from 'gatsby';

import SEO from '../SEO';

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: "Hi, I'm Kyle!",
          description: 'Personal website for Kyle Jones',
          author: 'Kyle Jones',
        },
      },
    })
  );
});

describe('SEO', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <SEO
          url="https://localhost/my-article"
          title="My Title"
          keywords={['my-keyword', 'my-second-keyword']}
          description="my-description"
          featuredImage="https://localhost/my-article-image.jpg"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
