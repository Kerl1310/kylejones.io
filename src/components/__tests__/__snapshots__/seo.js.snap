// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO renders correctly 1`] = `
<script
  data-schema="Author"
  type="application/ld+json"
>
   
  
          {
            "@context": "https://schema.org",
            "@type": "Author",
            "birthDate": "1992-10-13",
            "email": "kylejones1310@outlook.com",
            "gender": "Male",
            "jobTitle": "Senior Software Engineer",
            "knowsLanguage": "English",
            "name": "Kyle Jones",
            "telephone": "07432 829513",
            "url": "https://kylejones.io"
            "name": "Kyle Jones",
          }
        
</script>
`;
